src/components/ColourBar.vue

```diff
@@ -52,6 +80,36 @@ export default class ColourBar extends Vue {
       this.$vuetify.breakpoint.mobile ? 250 : 400, 30,
       options)
     this.isVisible = true
+
+    const firstChild = d3.select("#colourbar > g > g.axis").selectChild()
+    const lastChild = d3.select("#colourbar > g > g.axis").selectChild(':last-child')
+
+    this.range.min = firstChild.selectChild("text").property("innerHTML")
+    this.range.max = lastChild.selectChild("text").property("innerHTML")
+
+    firstChild.on('click', () => {
+      this.isEditingMin = true
+
+      this.$nextTick(() => {
+        const minInput = this.$refs.minInput as HTMLElement
+        minInput.focus()
+      })
+    })
+
+    lastChild.on('click', () => {
+      this.isEditingMax = true
+
+      this.$nextTick(() => {
+        const maxInput = this.$refs.maxInput as HTMLElement
+        maxInput.focus()
+      })
+    })
+  }
+
+  updatedRange() {
+    this.isEditingMin = false
+    this.isEditingMax = false
+    this.$emit("rangeUpdate", `${this.range.min},${this.range.max}`)
```

Comment:
Can we add some validation on the upper value of the mininput to be the maximput?